### PR TITLE
feat(gyro_odometer, imu_corrector): add roll and pitch rate

### PR DESCRIPTION
**Pull request by kminoda**
## Description(required)
I added roll and pitch rate in `sensing/imu_corrector` and `localization/gyro_odometer`. 
You can now provide roll and pitch rate (as well as yaw rate) to ekf_localizer. Note that the current ekf_localizer does not use roll and pitch rate.

## Review Procedure(required)
Change one of `angular_velocity_offset_x`, `angular_velocity_offset_y`, `angular_velocity_stddev_xx_`, or `angular_velocity_stddev_xx_` in your imu_corrector.param.yaml file. Make sure that the output of `imu_corrector` (`/sensing/imu/imu_data`) and `gyro_odometer` (e.g. `/localization/twist_estimator/twist_with_covariance`) reflect the change.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [commit-guidelines][commit-guidelines]
- [x] Assign PR to reviewer

If you are adding new package following items are required:

- [ ] Documentation with description of the package is available
- [ ] A sample launch file and parameter file are available if the package contains executable nodes

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[commit-guidelines]: https://www.conventionalcommits.org/en/v1.0.0/
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Review by YamatoAndo — APPROVED** [review]

**Comment by YamatoAndo** [comment]
It worked as expected!

**Comment by YamatoAndo** [comment]
@kminoda Could you add a description of the parameters to the readme?

**Review by YamatoAndo — CHANGES_REQUESTED** [review]
Could you add a description of the parameters to the readme?

**Comment by YamatoAndo** [comment]
@kminoda 
Apply the covariance of roll rate and pitch rate.

https://github.com/autowarefoundation/autoware.universe/blob/706e6e7ca4c0b04a673e3a1bf559d84d6d181617/localization/gyro_odometer/src/gyro_odometer_core.cpp#L105-L107

File: localization/gyro_odometer/src/gyro_odometer_core.cpp [hunk 1]
```diff
@@ -80,13 +82,17 @@ void GyroOdometer::callbackImu(const sensor_msgs::msg::Imu::ConstSharedPtr imu_m
...
   twist_pub_->publish(twist);
 
   geometry_msgs::msg::TwistWithCovarianceStamped twist_with_covariance;
   twist_with_covariance.header.stamp = imu_msg_ptr->header.stamp;
   twist_with_covariance.header.frame_id = output_frame_;
   twist_with_covariance.twist.twist.linear = twist_with_cov_msg_ptr_->twist.twist.linear;
+  twist_with_covariance.twist.twist.angular.x = transformed_angular_velocity.vector.x;
+  twist_with_covariance.twist.twist.angular.y = transformed_angular_velocity.vector.y;
   twist_with_covariance.twist.twist.angular.z =
     transformed_angular_velocity.vector.z;  // TODO(YamatoAndo) yaw_rate only
```
Comment (YamatoAndo) [inline localization/gyro_odometer/src/gyro_odometer_core.cpp:97]:
```suggestion
    transformed_angular_velocity.vector.z;
```

**Comment by YamatoAndo** [comment]
https://github.com/autowarefoundation/autoware.universe/blob/706e6e7ca4c0b04a673e3a1bf559d84d6d181617/localization/gyro_odometer/src/gyro_odometer_core.cpp#L99-L101

delete `angular.x, and angular.y` from this comment.

File: localization/gyro_odometer/src/gyro_odometer_core.cpp [hunk 2]
```diff
@@ -105,25 +107,28 @@ void GyroOdometer::callbackImu(const sensor_msgs::msg::Imu::ConstSharedPtr imu_m
...
-  // linear.y, linear.z, angular.x, and angular.y are not measured values.
+  // linear.y and linear.z are not measured values.
   // Therefore, they should be assigned large variance values.
   twist_with_covariance.twist.covariance[0] = twist_with_cov_msg_ptr_->twist.covariance[0];
   twist_with_covariance.twist.covariance[7] = 10000.0;
   twist_with_covariance.twist.covariance[14] = 10000.0;
-  twist_with_covariance.twist.covariance[21] = 10000.0;
-  twist_with_covariance.twist.covariance[28] = 10000.0;
-  twist_with_covariance.twist.covariance[35] = imu_msg_ptr_->angular_velocity_covariance[8];
+  twist_with_covariance.twist.covariance[21] = imu_msg_ptr->angular_velocity_covariance[0];
```
Comment (YamatoAndo) [inline localization/gyro_odometer/src/gyro_odometer_core.cpp:129]:
```suggestion
  twist_with_covariance.twist.covariance[21] = imu_msg_ptr_->angular_velocity_covariance[0];
```

File: localization/gyro_odometer/src/gyro_odometer_core.cpp [hunk 3]
```diff
@@ -105,25 +107,28 @@ void GyroOdometer::callbackImu(const sensor_msgs::msg::Imu::ConstSharedPtr imu_m
...
+  twist_with_covariance.twist.covariance[28] = imu_msg_ptr->angular_velocity_covariance[4];
```
Comment (YamatoAndo) [inline localization/gyro_odometer/src/gyro_odometer_core.cpp:130]:
```suggestion
  twist_with_covariance.twist.covariance[28] = imu_msg_ptr_->angular_velocity_covariance[4];
```

File: localization/gyro_odometer/src/gyro_odometer_core.cpp [hunk 4]
```diff
@@ -105,25 +107,28 @@ void GyroOdometer::callbackImu(const sensor_msgs::msg::Imu::ConstSharedPtr imu_m
...
+  twist_with_covariance.twist.covariance[35] = imu_msg_ptr->angular_velocity_covariance[8];
```
Comment (YamatoAndo) [inline localization/gyro_odometer/src/gyro_odometer_core.cpp:131]:
```suggestion
  twist_with_covariance.twist.covariance[35] = imu_msg_ptr_->angular_velocity_covariance[8];
```

File: localization/gyro_odometer/src/gyro_odometer_core.cpp (hunk 2 above)
Comment (kminoda) [reply localization/gyro_odometer/src/gyro_odometer_core.cpp:129]: Why do you suggest adding the trailing underscore?

Comment (YamatoAndo) [reply localization/gyro_odometer/src/gyro_odometer_core.cpp:129]: The underscores are attached in other parts of the code, so I think it is better to do the same.

**Review by yukkysaito — APPROVED** [review]

**Review by YamatoAndo — APPROVED** [review]

**Pull request merged**